napari/layers/labels/_labels_key_bindings.py

```diff
@@ -153,4 +158,5 @@ def reset_polygon(layer: Labels):
 )
 def complete_polygon(layer: Labels):
     """Complete the drawing of the current polygon."""
-    layer._overlays["polygon"].add_polygon_to_labels(layer)
+    polygon_overlay = cast('LabelsPolygonOverlay', layer._overlays["polygon"])
+    polygon_overlay.add_polygon_to_labels(layer)
```

Comment:
`layer._overlays` is defined as an EventedDict containing `Overlay`s in values on the base class.

add_polygon_to_labels is only defined on the subclass LabelsPolygonOverlay

We could use `typing.TypedDict` but those are not typed, we could have a `TypedEventedDict`.

So this is really just to please mypy.

We could open an issue for EventedTypedDict, or just have a custom object as we don't have that many keys for this dict. 

```
'polygon'
'axes'
'bounding_box'
'brush_circle'
'dot'
'scale_bar'
'text'
'transform_box'
```

Reply:
I *really* don't like having a hard to read, more verbose, and more complicated code just to make mypy happy and not because it's actually useful to have types there :/ can't we just ignore this?

Reply:
I guess we could use  `type: ignore`, the issue with ignore is that it may hide other issues on the given line. I don't have much preferences.

Reply:
👍 I added a type ignore and a comment with TODO so hopefully someone can look at this again in the future